Packs/PcapAnalysis/Scripts/PcapFileExtractor/PcapFileExtractor.py

```diff
@@ -0,0 +1,195 @@
+            protocol = 'dicom'
+
+        if line.endswith(')') and 'TFTP' in line:
+            data_list.append(line)
+            protocol = 'tftp'
+
+    return protocol, data_list
+
+
+def extract_files(file_path, dir_path):
+    """
+    :param file_path: the pcap file path.
+    :param dir_path: directory for the files will be extract to
+    :return:
+        excluded_files: the excludedfiles which are in dir_path
+    """
+    protocol, _ = find_files_protocol(file_path)
+    if not protocol:
+        return_error('Could not find a valid protocol for extracting the files')
+
+    if protocol == 'http':
+        command_ = f'tshark -r {file_path} --export-objects http,{dir_path}'
+
+    elif protocol == 'smb':
+        command_ = f'tshark -r {file_path} --export-objects smb,{dir_path}'
+
+    elif protocol == 'imf':
+        command_ = f'tshark -r {file_path} --export-objects imf,{dir_path}'
+
+    elif protocol == 'tftp':
+        command_ = f'tshark -r {file_path} --export-objects tftp,{dir_path}'
+
+    elif protocol == 'dicom':
+        command_ = f'tshark -r {file_path} --export-objects dicom,{dir_path}'
+
+    process = subprocess.Popen(command_, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    process.communicate()
+    excluded_files = [f for f in glob.glob(dir_path + '/*')]
+
+    return excluded_files
+
+
+def upload_files(excluded_files, dir_path, file_path):
+    """
+
+    :param excluded_files: excluded files
+    :param dir_path: dir path for the files
+    :param file_path: the path to the pcap file
+    :return:
+    """
+    filenames = []  # type: ignore
+    # recursive call over the file system top down
+    for root, directories, files in os.walk(dir_path):
+        for f in files:
+            # skipping previously existing files
+            # adding it to the extracted pcap files list
+            if f not in excluded_files and isfile(os.path.join(root, f)):
+                filenames.append(os.path.join(root, f))
+
+    if len(filenames) == 0:
+        return_error('Could not find files')
+
+    else:
+        results = []
+        context = []
+        protocol, packet_data = find_files_protocol(file_path)
+
+        md5 = hashlib.md5()
+        sha1 = hashlib.sha1()
+        sha256 = hashlib.sha256()
+        files_base_names = [os.path.basename(file_path) for file_path in filenames]  # noqa[F812]
+        files_dic = {file_path: os.path.basename(file_path) for file_path in filenames}
+
+        for file_path, file_name in files_dic.items():
+            for data in packet_data:
+                packet_number = data.split()[0]
+                for packet_number in packet_data:
+                    data = [i for i in packet_number.split()]
+                    source_ip = data[2]
+                    dest_ip = data[4]
+
+            with open(file_path, 'rb') as _file:
+                demisto.results(fileResult(file_name, _file.read()))
+
+            with open(file_path, 'rb') as _file:
```

Comment:
why are you opening the file twice?

Reply:
Tried to do it in the same opening of the file, but it doesn't work. 
Seems to lose data in the process

Reply:
you cant call `read()` twice, call it once and store in variable to use